File: backend/InvoiceManage/InvoiceManage/src/main/java/com/example/InvoiceManage/entity/InvoiceRequest.java
package com.example.InvoiceManage.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "invoice_request")
@Getter
@Setter
public class InvoiceRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user; // người yêu cầu

    @ManyToOne
    @JoinColumn(name = "order_id", nullable = false)
    private Order order; // đơn hàng muốn hóa đơn

    @ManyToOne
    @JoinColumn(name = "status_id", nullable = false)
    private Status status; // pending / accepted / rejected

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

}

File: base_vite_ts/src/types/InvoiceRequest.ts
export interface InvoiceRequest {
    id: number;
    userId: number;
    orderId: number;
    statusId: number;
    createdAt: string;
}

export const INVOICE_STATUS_OPTIONS = [
    { id: 1, value: 'pending', label: 'Chờ xét duyệt' },
    { id: 2, value: 'rejected', label: 'Từ chối' },
    { id: 3, value: 'awaiting_payment', label: 'Chờ thanh toán' },
    { id: 4, value: 'paid', label: 'Đã thanh toán' },
]; 